currency_exchange/app.py

```diff
@@ -2,47 +2,42 @@
 import argparse
 import requests
 
-
 CURRENCIES = {
     'EUR': 4.75,
     'USD': 4.80,
     'CAD': 3.50
 }
 
+
 def parser():
     parser = argparse.ArgumentParser(description='Exchange the currencies')
 
     parser.add_argument('amount', type=int,
                         help='an integer for the currency')
-    parser.add_argument('currency', type=str, 
-        choices=CURRENCIES.keys(), help='currency type')
-
+    parser.add_argument('currency', type=str,
+                        choices=CURRENCIES.keys(), help='currency type')
 
     args = parser.parse_args()
 
     return args
 
+
 args = parser()
 
+
 def fetch_currency_rate(currency: str) -> float:
-    r = requests.get('http://api.nbp.pl/api/exchangerates/tables/A/')
+    r = requests.get(f'http://api.nbp.pl/api/exchangerates/rates/A/{currency}/')
 
-    rates = r.json()[0]["rates"]
+    rates = r.json()
+    if rates['code'] == currency:
+        return rates['rates'][0]['mid']
 
-    for i in rates:
-        if i["code"] == currency:
-            return i["mid"]
 
 def convert(value: float, currency: str) -> float:
     exchange_rate = fetch_currency_rate(currency)
-    return value * exchange_rate
-    # return exchange_rate * value
-
+    return round(value * exchange_rate, 2)
 
 
 input = (args.amount, args.currency)
```

Comment:
Please move `args = parser()` from line 25 to the closest line up to `input = (args.amount, args.currency)` (currently line 40)

Reply:
Done